File: ruby/1-25/problem_3.rb
=begin
                            Largest prime factor

    The prime factors of 13195 are 5, 7, 13 and 29.

    What is the largest prime factor of the number 600851475143 ?

                            Answer: 6857
=end

#!/usr/bin/ruby

require_relative '../project_euler'

def largest_prime( max )
    return prime_factors( max ).sort.pop
end

puts largest_prime( 600851475143 );

File: src/python/1-25/problem_3.py
"""
                            Largest prime factor

    The prime factors of 13195 are 5, 7, 13 and 29.

    What is the largest prime factor of the number 600851475143 ?

                            Answer: 6857
"""
import sys
sys.path.append('../')
# pylint: disable=wrong-import-position,import-error
import project_euler as pe

# pylint: disable=redefined-builtin
def largest_prime(max):
    """Given max this function returns the largest prime factor"""
    return pe.prime_factors(max)[-1]

print(largest_prime(600851475143))
